lambdas/data-migration2/src/executions.ts

```diff
@@ -0,0 +1,99 @@
+import Knex from 'knex';
+import Logger from '@cumulus/logger';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import { envUtils } from '@cumulus/common';
+import { ExecutionRecord } from '@cumulus/types/api/executions';
+import { PostgresExecutionRecord, translateApiExecutionToPostgresExecution } from '@cumulus/db';
+import { RecordAlreadyMigrated } from './errors';
+import { MigrationSummary } from './types';
+
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+
+const logger = new Logger({ sender: '@cumulus/data-migration/executions' });
+
+/**
+ * Migrate execution record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Source record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordAlreadyMigrated}
+ *   if record was already migrated
+ */
+export const migrateExecutionRecord = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<void> => {
+  // Use API model schema to validate record before processing
+  Manager.recordIsValid(dynamoRecord, schemas.execution);
+
+  const existingRecord = await knex<PostgresExecutionRecord>('executions')
+    .where({
+      arn: dynamoRecord.arn,
+    })
+    .first();
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`Execution arn ${dynamoRecord.arn} was already migrated, skipping`);
+  }
+
+  const updatedRecord = await translateApiExecutionToPostgresExecution(
+    <ExecutionRecord>dynamoRecord, <Knex>knex
+  );
+
+  if (updatedRecord.parent_cumulus_id !== undefined) {
+    // Get parent record
+    // Migrate parent record
+  }
```

Comment:
I originally had this logic written in db/src/executions.ts but I don't think that's right. Workin on moving it here.